_docs/use_functions.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Using Functions
+category: Common Concepts
+---
+
+You've already encountered the `main` function, the entrypoint of any Glu program, as well as the `func` keyword which is needed to declare a function.
+
+Procedural languages, like Glu, structure programs as a series of functions that call one another.
+This is why functions are essential in Glu programming.
+
+```glu
+func main() {
+    std::print("Hello World!");
+}
+```
+
+## Declare a Function
+
+To declare a function, four key components are required:
+- The `func` keyword
+- The function name
+- A list of any necessary parameters
+- The function potential returned value's type
+
+The body of the function is enclosed in curly brackets `{}`, which indicate to the compiler where the function begins and ends.
+
+In the example below, the function prototype of `my_func` specifies the function name and parameters: `param1` and `param2`, both of which are of type `Int`.
+
+![Representation of the differents key components required by a function declaration in Glu](/assets/img/function_explaining.png)
+
+A function can be declared without being immediately implemented.
+In such cases, a semicolon `;` is used instead of curly brackets.
+Although functions do not always require parameters, the `func` keyword and the function name are always necessary for declaration.
+
+## Function Parameters
+
+A function may not always require parameters, but what exactly is a parameter ?
+Parameters are special variables that appear in a function's signature.
+When a function requires parameters, they must be provided with concrete values.
+
+In the following version of `my_func`, the parameters `param1` and `param2` are assigned the values `42` and `42`:
+
+```glu
+func main() {
+    my_func(42, 42); // This will print 84
+}
+
+func my_func(param1: Int, param2: Int) {
+    std::print(param1 + param2);
+}
+```
+
+This function will print `84` as `my_func` simply adds `param1` to `param2`.
+
+But, optional parameters can be specified in Glu by specifying default values to any function parameter:
+
+```glu
+func main() {
+    my_func(2); // This will print 44
+}
+
+func my_func(param1: Int = 42, param2: Int) {
```

Comment:
No, just like C++, optional parameters must be at the end, otherwise it is ambiguous (if both are optional, which one is it we’re supplying?)